src/AtelierTomato.Markov.Core/SentenceParser.cs

```diff
@@ -18,7 +18,7 @@ public class SentenceParser
 |                        # nor
 [.!?]\w                  # punctuation with word after it
 )+)                      # 1 or multiple
-(\.|[.!?]+|$|\r?\n)", RegexOptions.IgnorePatternWhitespace | RegexOptions.Compiled);
+((\.|[.!?]+)[\)\]}""”«»]*|(|$|\r?\n))", RegexOptions.IgnorePatternWhitespace | RegexOptions.Compiled);

```

Comment:
the regex is a bit sus, with the weird preceding alternation operator before the end-of-string and the extra groups that are not non-capturing
i think this would be more in line with how the regex is written, although an explanation comment would be nice to explain it
̀ ((?:\.|[.!?]+)[\)\]}""”«»]*|(?:$|\r?\n))`